ci/Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+            XGBOOST_REPO="rapidsai/xgboost"
+            XGBOOST_BRANCH="branch-22.08"
+            XGBOOST_CONDA_BRANCH="branch-22.08"
+            RAPIDS_VER="22.08"
+        }
+
+        matrix {
+            agent any
+            axes {
+                axis {
+                    name 'PYTHON'
+                    values '3.8', '3.9'
+                }
+                axis {
+                    name 'CUDA'
+                    values '11.5'
+                }
+                axis {
+                    name 'ARCH'
+                    values 'amd64', 'arm64'
+                }
+            }
+            stages {
+                stage("Build PR") {
+                    when { environment name: 'BUILD_TYPE', value: 'pull-request' }
+                    agent {
+                        label "cpu4-${ARCH}"
+                    }
+                    environment {
+                        BUILD_TAG="xgboost-conda-build-$CUDA-$PYTHON"
+                        CONDA_USERNAME="${CONDA_CHANNEL}"
+                    }
+
+                    steps {
+                        script {
+                            stage("Build PR: $CUDA - $PYTHON - $ARCH") {
+                                def image = getImg("$PYTHON", "$ARCH", "$CUDA")
+                                docker.image("${image}").inside("""
+                                --pull=always
+                                -e CUDA=${CUDA}
+                                -e PY_VER=${PYTHON}
+                                """) {
+                                    sh "${buildString}"
+                                }
+                            }
+                        }
+                    }
+                }
+                stage("Build Branch") {
+                    when { environment name: 'BUILD_TYPE', value: 'nightly' }
+                    agent {
+                        label "cpu4-${ARCH}"
+                    }
+                    environment {
+                        BUILD_TAG="xgboost-conda-build-$CUDA-$PYTHON"
+                        CONDA_USERNAME="${CONDA_CHANNEL}"
+                        MY_UPLOAD_KEY=credentials("rapidsai-nightly-conda-upload-api")
+                        RAPIDSAI_CONDA_KEY=credentials("rapidsai-conda-upload-api")
+                        NVIDIA_UPLOAD_KEY=credentials("nvidia-conda-upload-key")
+                    }
+
+                    steps {
+                        script {
+                            stage("Build Branch: $CUDA - $PYTHON - $ARCH") {
+                                def image = getImg("$PYTHON", "$ARCH", "$CUDA")
+                                docker.image("${image}").inside("""
+                                --pull=always
+                                -e CUDA=${CUDA}
+                                -e PY_VER=${PYTHON}
+                                """) {
+                                    sh "${buildString}"
+                                }
+                            }
+                        }
```

Comment:
In previous pipelines, we only used the `script` step to enable us to run an array of build steps from the shared-library in parallel.

Since we're using the `matrix` directive here, we can probably skip the `script` step and just use the default Docker functionality provided by Jenkins. described in the `docker` section below

https://www.jenkins.io/doc/book/pipeline/syntax/#agent